scene/gui/tab_bar.cpp

```diff
@@ -289,6 +297,36 @@ void TabBar::gui_input(const Ref<InputEvent> &p_event) {
 			}
 		}
 	}
+
+	if (p_event->is_pressed()) {
+		Input *input = Input::get_singleton();
+		Ref<InputEventJoypadMotion> joypadmotion_event = p_event;
+		Ref<InputEventJoypadButton> joypadbutton_event = p_event;
+		bool is_joypad_event = (joypadmotion_event.is_valid() || joypadbutton_event.is_valid());
+		if (p_event->is_action("ui_right", true)) {
+			if (is_joypad_event) {
+				if (!input->is_action_just_pressed("ui_right", true)) {
+					return;
+				}
+				set_process_internal(true);
+			}
+			bool success = select_tab_on_right();
+			if (success) {
+				accept_event();
+			}
+		} else if (p_event->is_action("ui_left", true)) {
+			if (is_joypad_event) {
+				if (!input->is_action_just_pressed("ui_left", true)) {
+					return;
+				}
+				set_process_internal(true);
+			}
+			bool success = select_tab_on_left();
+			if (success) {
+				accept_event();
+			}
```

Comment:
The `success` variables seem pointless since they are only used once right below. I think the functions should be used directly instead.

Reply:
This is more of my personal style to avoid calling functions affecting state inside the "if" test. But sure I can change it.